dll/win32/shell32/shelldesktop/CDirectoryWatcher.cpp

```diff
@@ -385,10 +384,17 @@ void CDirectoryWatcher::QuitWatching()
     assert(this != NULL);
 
     m_fDead = TRUE;
+    m_hNotifyWnd = NULL;
     CancelIo(m_hDirectory);
 }
 
-BOOL CDirectoryWatcher::IsDead() const
+BOOL CDirectoryWatcher::IsDead()
 {
+    if (m_hNotifyWnd && !::IsWindow(m_hNotifyWnd))
```

Comment:
Does this happen?
If so, that seems like a logic error somewhere else?

Reply:
The watcher's side is not system but application side. Bad application can destroy a window that is watching.